Integrated-Windows-Auth-Flow/README.md

```diff
@@ -0,0 +1,174 @@
+   Change your portal session to the desired Azure AD tenant.
+1. In the portal menu, select the **Azure Active Directory** service, and then select **App registrations**.
+
+> In the next steps, you might need the tenant name (or directory name) or the tenant ID (or directory ID). These are presented in the **Properties** of the Azure Active Directory window respectively as *Name* and *Directory ID*
+
+#### Register the client app (iwa-console)
+
+1. Navigate to the Microsoft identity platform for developers [App registrations](https://go.microsoft.com/fwlink/?linkid=2083908) page.
+1. Select **New registration**.
+   - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `iwa-console`.
+   - In the **Supported account types** section, select **Accounts in any organizational directory**.
+   - Click **Register** button at the bottom to create the application.
+1. On the application **Overview** page, find the **Application (client) ID** and **Directory (tenant) ID** values and record it for later. You'll need it to configure the configuration file(s) later in your code.
+1. In the list of pages for the app, select **Manifest**, and:
+   - In the manifest editor, set the **allowPublicClient** property to **true**
+   - Select **Save** in the bar above the manifest editor.
+1. In the list of pages for the app, select **API permissions**
+   - Click the **Add a permission** button and then,
+   - Ensure that the **Microsoft APIs** tab is selected
+   - In the *Commonly used Microsoft APIs* section, click on **Microsoft Graph**
+   - In the **Delegated permissions** section, ensure that the right permissions are checked: **User.Read**, **User.ReadBasic.All**. Use the search box if necessary.
+   - Select the **Add permissions** button
+
+1. At this stage permissions are assigned correctly but the client app does not allow interaction. Therefore no consent can be presented via a UI and accepted to use the service app.
+   Click the **Grant/revoke admin consent for {tenant}** button, and then select **Yes** when you are asked if you want to grant consent for the requested permissions for all account in the tenant.
+   You need to be an Azure AD tenant admin to do this.
+
+### Step 4:  Configure the sample to use your Azure AD tenant
+
+In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+Open the `IntegratedWindowsAuthFlow`class to configure the project.
+
+#### Configure the client project
+
+1. Open the `src\main\java\IntegratedWindowsAuthFlow` class.
+1. Find the line where `PUBLIC_CLIENT_ID` is set and replace the existing value with the `Application ID (clientId)` of the `iwa-console` application copied from the Azure portal.
+1. Find the line where `USER_NAME` is set and replace the existing value with your `Windows signed-in user ID`.
```

Comment:
@kalyankrishna1 I don't think this step is necessary, right? It looks like the code is asking user to Enter the username and then capturing it in a variable USER_NAME which is used.